zazu/dev/commands.py

```diff
@@ -193,58 +196,56 @@ def status(ctx):
             except zazu.issue_tracker.IssueTrackerError:
                 click.echo("    No ticket found")
 
             matches = pulls_future.result()
-            click.secho('Pull request info:', bg='white', fg='black')
-            click.echo('    {} matching PRs'.format(len(matches)))
+            click.secho('Review info:', bg='white', fg='black')
+            click.echo('    {} matching reviews'.format(len(matches)))
             if matches:
                 for p in matches:
-                    click.echo(click.style('    PR Name:  ', fg='green') + p.title)
-                    click.echo(click.style('    PR State: ', fg='green') + p.state)
-                    click.echo(click.style('    PR Body:\n', fg='green') + wrap_text(p.body))
+                    click.echo('{} {}'.format(click.style('    Review: '.format(type.capitalize()), fg='green'), p.name))
+                    click.echo('{} {}, {}'.format(click.style('    Status:', fg='green'), p.status, 'merged' if p.merged else 'unmerged'))
+                    click.echo('{} {} -> {}'.format(click.style('    Branches:', fg='green'), p.head, p.base))
+                    click.echo(click.style('    Description:\n', fg='green') + wrap_text(p.description))
 
                     # TODO: build status from TC
 
 
 @dev.command()
 @click.pass_context
-def review(ctx):
+@click.option('--base', help='The base branch to target')
+@click.option('--head', help='The head branch (defaults to current branch and origin organization)')
+def review(ctx, base, head):
     """Create or display pull request"""
-    import urllib
-    encoded_branch = urllib.quote_plus(ctx.obj.repo.active_branch.name)
-    url = ctx.obj.repo.remotes.origin.url
-    start = 'github.com'
-    if start in url:
-        base_url = url[url.find(start):].replace('.git', '').replace(':', '/')
-        url = 'https://{}/compare/{}?expand=1'.format(base_url, encoded_branch)
-        click.echo('Opening "{}"'.format(url))
-        webbrowser.open_new(url)
-        # TODO: add link to ticket in the PR, zazu logo
-        # <img src="http://vignette1.wikia.nocookie.net/disney/images/c/ca/Zazu01cf.png" alt="Zazu" width=50"/>
-    else:
-        raise click.UsageError("Can't open a PR for a non-github repo")
+    code_reviewer = ctx.obj.code_reviewer()
+    head = ctx.obj.repo.active_branch.name if head is None else head
+    existing_reviews = code_reviewer.review(status='open', head=head, base=base)
+    try:
+        pr = zazu.util.pick(existing_reviews, 'Multiple reviews found, pick one')
+    except IndexError:
+        base = 'develop' if base is None else base
+        click.echo('No existing review found, creating one...')
+        descriptor = make_issue_descriptor(ctx.obj.repo.active_branch.name)
+        issue_id = descriptor.id
+        title = zazu.util.prompt('Title', default=descriptor.readable_description())
+        body = '{}\n\nFixes #{}'.format(zazu.util.prompt('Summary'), issue_id)
```

Comment:
So this relied on the parsing logic in make_issue_descriptor() but you bring up a good point. If the branch was not properly named we got a uninteligible exception thrown. This much more robust after some tweaking, great catch.